modules/code-deployment/src/main/java/org/apache/ignite/internal/deployunit/DeploymentManagerImpl.java

```diff
@@ -0,0 +1,424 @@
+        if (id == null || id.isBlank()) {
+            return CompletableFuture.failedFuture(new DeploymentUnitIdentifierException());
+        }
+        CompletableFuture<List<Version>> result = new CompletableFuture<>();
+        metaStorage.prefix(new ByteArray(UNITS_PREFIX + id))
+                .subscribe(new Subscriber<>() {
+                    private final List<Version> set = new ArrayList<>();
+
+                    @Override
+                    public void onSubscribe(Subscription subscription) {
+                        subscription.request(Long.MAX_VALUE);
+                    }
+
+                    @Override
+                    public void onNext(Entry item) {
+                        UnitMeta deserialize = UnitMetaSerializer.deserialize(item.value());
+                        set.add(deserialize.getVersion());
+                    }
+
+                    @Override
+                    public void onError(Throwable throwable) {
+                        result.completeExceptionally(throwable);
+                    }
+
+                    @Override
+                    public void onComplete() {
+                        result.complete(set);
+                    }
+                });
+        return result;
+    }
+
+    @Override
+    public CompletableFuture<UnitStatus> statusAsync(String id) {
+        if (id == null || id.isBlank()) {
+            return CompletableFuture.failedFuture(new DeploymentUnitIdentifierException());
+        }
+        CompletableFuture<UnitStatus> result = new CompletableFuture<>();
+        metaStorage.prefix(new ByteArray(UNITS_PREFIX + id))
+                .subscribe(new Subscriber<>() {
+                    private UnitStatusBuilder builder;
+
+                    @Override
+                    public void onSubscribe(Subscription subscription) {
+                        subscription.request(Long.MAX_VALUE);
+                    }
+
+                    @Override
+                    public void onNext(Entry item) {
+                        if (builder == null) {
+                            builder = UnitStatus.builder(id);
+                        }
+                        UnitMeta deserialize = UnitMetaSerializer.deserialize(item.value());
+                        builder.append(deserialize.getVersion(), deserialize.getConsistentIdLocation());
+                    }
+
+                    @Override
+                    public void onError(Throwable throwable) {
+                        result.completeExceptionally(throwable);
+                    }
+
+                    @Override
+                    public void onComplete() {
+                        if (builder != null) {
+                            result.complete(builder.build());
+                        } else {
+                            result.completeExceptionally(
+                                    new DeploymentUnitNotExistException("Unit with " + id + "not exist."));
```

Comment:
First of all, there should be a space before "not exist."
Second, "not exists" is grammatically incorrect, it should be "doesn't exist".